lib/src/rules/type_annotate_public_apis.dart

```diff
@@ -119,6 +122,9 @@ class _Visitor extends SimpleAstVisitor<void> {
       node.variables.accept(v);
     }
   }
+
+  bool _isParentClassOrMixinPrivate(ClassMember node) =>
+      isPrivate((node.parent as ClassOrMixinDeclaration).name);
```

Comment:
This cast looks iffy but maybe it's reasonable not to check first...  @bwilkerson?

Reply:
Yeah I don't think any of the nodes in the tree override `parent` to specify a tighter type. How is this typically handled?

Reply:
The question is: will a `ClassMember` ever be a child of something other than a `ClassOrMixinDeclaration`? I don't know. I don't believe that can happen today, but 3 months ago it was true that the parent was always a `ClassDeclaration` and then we added mixins to the language. Who knows what we'll add in the future.

The bottom line is that we can't make this completely future proof. Given that, maybe it's better for the method to fail when our expectations are violated than to have the method silently return the wrong answer.

Reply:
SG. When you say, "maybe it's better for the method to fail..." do you mean maybe its better to leave the method as it is?